lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -328,15 +328,16 @@ private function getShortName(string $className): string
     private function addInheritedFields(ClassMetadata $subClass, ClassMetadata $parentClass): void
     {
         foreach ($parentClass->fieldMappings as $mapping) {
-            if (! isset($mapping['inherited']) && ! $parentClass->isMappedSuperclass) {
-                $mapping['inherited'] = $parentClass->name;
+            $subClassMapping = clone $mapping;
```

Comment:
Lost a huge amount of time on this, but yeah, we have to clone here now.

Reply:
#10397 changed that area, did you include/merge that already?

why can’t you just update the value?

Reply:
I need to update the child value without updating the parent value. When using an array, a copy is made. When using objects, we're altering the parent objects.

We do need to do a merge up.

Reply:
Would it be better to do the clone up in `doLoadMetadata` or so, in a more central location?

Reply:
Merge up done in https://github.com/doctrine/orm/pull/10421

About `doLoadMetadata`, I'm not sure… I mean there's one clone per mapping to do, so it's not like you could pass clones to `addInheritedFields` or something :thinking: